ext/objspace/objspace.c

```diff
@@ -38,10 +38,14 @@
  *  information as only a *HINT*. Especially, the size of +T_DATA+ may not be
  *  correct.
  *
- *  This method is only expected to work with C Ruby.
- *
- *  From Ruby 2.2, memsize_of(obj) returns a memory size includes
- *  sizeof(RVALUE).
+ *  This method is only expected to work with CRuby.
+ *
+ *  From Ruby 2.2, memsize_of(obj) returns a memory size that includes
+ *  the slot size occupied by the object. Prior to Ruby 3.2, this was a
+ *  fixed size (sizeof(RVALUE)). From Ruby 3.2 with Variable Width
+ *  Allocation, this is the actual slot size used (1x, 2x, 4x, 8x, or
+ *  16x BASE_SLOT_SIZE), plus any additional memory allocated outside
+ *  the slot (such as external strings, arrays, or hash tables).
```

Comment:
We're only documenting for the latest version of Ruby, so I don't think we should include info about prior versions.